src/router.rs

```diff
@@ -73,10 +73,15 @@ impl Router {
     /// a controller function, so that you can confirm that the request is
     /// authorized for this route before handling it.
     pub fn route<S: AsRef<str>, H: Handler, I: AsRef<str>>(&mut self, method: method::Method, glob: S, handler: H, route_id: I) -> &mut Router {
-        self.mut_inner().routers
-            .entry(method)
-            .or_insert(Recognizer::new())
-            .add(glob.as_ref(), Box::new(handler));
+
+        let mut hash: HashMap<method::Method, Arc<Handler>> = HashMap::new();
+
+        if let Some(s) = self.mut_inner().routers.recognize(glob.as_ref()).ok() {
+            hash = s.handler.clone();
+        }
+
+        hash.insert(method, Arc::new(handler));
+        self.mut_inner().routers.add(glob.as_ref(), hash);
```

Comment:
I'm not sure how you would avoid cloning the hashmap and adding it again, an API in Recognizer to get a `&mut HashMap` instead of `&HashMap` would help.